### PR TITLE
storage: first version of proto

### PR DESCRIPTION
**Pull request by glimchb**
Starting to implement https://github.com/opiproject/opi-api/blob/main/storage/spec.md

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
server
```
$ docker run --rm -it -v `pwd`:`pwd` -w `pwd` -p 50051:50051 golang:alpine go run server.go
go: downloading google.golang.org/grpc v1.48.0
go: downloading google.golang.org/protobuf v1.28.0
go: downloading google.golang.org/genproto v0.0.0-20220725144611-272f38e5d71b
go: downloading github.com/golang/protobuf v1.5.2
go: downloading golang.org/x/net v0.0.0-20220725212005-46097bf591d3
go: downloading golang.org/x/sys v0.0.0-20220722155257-8c9f86f7a55f
go: downloading golang.org/x/text v0.3.7
2022/07/26 02:48:20 server listening at [::]:50051
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
2022/07/26 02:48:56 Received: OPI-Nvme
^Csignal: interrupt
```
client
```
 $ docker run --net=host --rm -it -v  `pwd`:`pwd` -w `pwd` golang:alpine go run client.go
go: downloading google.golang.org/grpc v1.48.0
go: downloading google.golang.org/protobuf v1.28.0
go: downloading github.com/golang/protobuf v1.5.2
go: downloading google.golang.org/genproto v0.0.0-20220725144611-272f38e5d71b
go: downloading golang.org/x/net v0.0.0-20220725212005-46097bf591d3
go: downloading golang.org/x/sys v0.0.0-20220722155257-8c9f86f7a55f
go: downloading golang.org/x/text v0.3.7
2022/07/26 02:48:56 Added: Hello OPI-Nvme
2022/07/26 02:48:56 Deleted: Hello OPI-Nvme
2022/07/26 02:48:56 Got: Hello OPI-Nvme
2022/07/26 02:48:56 Added: Hello OPI-Nvme
2022/07/26 02:48:56 Deleted: Hello OPI-Nvme
2022/07/26 02:48:56 Got: Hello OPI-Nvme
2022/07/26 02:48:56 Added: Hello OPI-Nvme
2022/07/26 02:48:56 Deleted: Hello OPI-Nvme
2022/07/26 02:48:56 Got: Hello OPI-Nvme
2022/07/26 02:48:56 Disconnected: Hello OPI-Nvme
```

**Review by mestery — CHANGES_REQUESTED** [review]
This needs a `README.md` file which states explicitly the following:
* This directory contains the OPI Storage API
* The client/server are meant as example implementations and are not meant for production use

It also needs CI infra to build and test this out, likely including unit tests. This part we can do in a follow-on (very quickly), but the `README.md` should happen before merge.

File: storage/proto/backend.proto [hunk 1]
```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
```
Comment (seroyer) [inline storage/proto/backend.proto:1]: Is this short term until we prove it works or long term home for the pb?  I would have imagined that one would find the API pb in the opi-api repo.

Comment (glimchb) [reply storage/proto/backend.proto:1]: Yes, short term
I will move `proto` folder it to near the spec here https://github.com/opiproject/opi-api/blob/main/storage/spec.md

Comment (mestery) [reply storage/proto/backend.proto:1]: Yes, long-term, this does indeed move out to the `opi-api` repo.

**Review by mestery — APPROVED** [review]
Fine as a first step.

**Review by seroyer — APPROVED** [review]

**Comment by glimchb** [comment]
> This needs a `README.md` file which states explicitly the following:
> 
> * This directory contains the OPI Storage API
> * The client/server are meant as example implementations and are not meant for production use
> 
> It also needs CI infra to build and test this out, likely including unit tests. This part we can do in a follow-on (very quickly), but the `README.md` should happen before merge.

done

**Pull request merged**